reflex/utils/processes.py

```diff
@@ -126,21 +126,21 @@ def handle_port(service_name: str, port: int, default_port: int) -> int:
     Args:
         service_name: The frontend or backend.
         port: The provided port.
-        default_port: The default port number associated with the specified service.
+        autoswitch: Whether to automatically switch the port.
 
     Returns:
         The port to run the service on.
 
     Raises:
         Exit:when the port is in use.
     """
-    if is_process_on_port(port):
-        if port == int(default_port):
-            return change_port(port, service_name)
-        else:
-            console.error(f"{service_name.capitalize()} port: {port} is already in use")
-            raise typer.Exit()
-    return port
+    if not is_process_on_port(port):
+        return port
+    if autoswitch:
+        return change_port(port, service_name)
+    else:
+        console.error(f"{service_name.capitalize()} port: {port} is already in use")
```

Comment:
if we could get the pid from `is_process_on_port`, that might improve this error message